### PR TITLE
swap: use p2ptest for protocol tests

### PR DESCRIPTION
**Pull request by ralph-pichler**
This PR
* rewrites `TestEmitCheque` to use `p2ptest` to also verify that the correct message is being sent
* rewrites `TestTriggerPaymentThreshold` to use `p2ptest` to also verify that the correct message is being sent
* updates `TestTriggerDisconnectThreshold` to test the right thing for pending cheques

### CODE REVIEW

**Review by zelig — APPROVED** [review]

File: swap/protocol_test.go [hunk 1]
```diff
@@ -217,61 +215,60 @@ func TestHandshakeInvalidContract(t *testing.T) {
...
+// We send a EmitChequeMsg to the creditor which handles the cheque and sends a ConfirmChequeMsg
 func TestEmitCheque(t *testing.T) {
 	testBackend := newTestBackend()
-	defer testBackend.Close()
+	protocolTester, clean, err := newSwapTester(t, testBackend)
+	defer clean()
+	if err != nil {
+		t.Fatal(err)
+	}
+	swap := protocolTester.swap
```
Comment (holisticode) [inline swap/protocol_test.go]: Can we please keep the `creditorSwap` terminology?

It is easier then to reason which swap it is when it stands on its own

File: swap/protocol_test.go [hunk 2]
```diff
@@ -379,13 +392,44 @@ func TestTriggerPaymentThreshold(t *testing.T) {
...
-	if creditor.getLastSentCheque() == nil {
-		t.Fatal("Expected one cheque, but there is none")
+	// we wait until the confirm message has been processed
+loop:
+	for {
+		select {
+		case <-ctx.Done():
+			t.Fatal("Expected one cheque, but there is none")
+		default:
+			if creditor.getLastSentCheque() != nil {
```
Comment (holisticode) [inline swap/protocol_test.go:428]: This actually means that if the remote peer does not confirm the cheque, the node remains "stuck" (`lastSentCheque` would not be released). So if a new cheque would be due and for some reason the confirmation msg never arrived, can the node process it based on the new cumulative amount or something? Or will it reject it

File: swap/protocol_test.go (hunk 1 above)
Comment (ralph-pichler) [reply swap/protocol_test.go]: Makes sense. I changed the name back to `creditorSwap `.

File: swap/protocol_test.go (hunk 2 above)
Comment (ralph-pichler) [reply swap/protocol_test.go:428]: As long as there is no confirmation the node will try to resend the current pending cheque once the threshold is reached again (the recipient will send the confirm message even if it has already processed it in the past) and then afterwards send a new cheque for the amount of debt incurred in the meantime. One of the reasons why we don't want to send a new cheque with a higher amount before that is that if the old cheque never was processed correctly by the recipient, then the recipient will consider the difference in `cumulativePayout` compared to the last cheque it has seen as the payment for the honey amount specified in the latest cheque (which doesn't match the expected price and will therefore be rejected). If there is no honey-to-money anymore in the new swap pricing, maybe we can also allow trying to resend with a higher payout.

File: swap/protocol_test.go (hunk 1 above)
Comment (holisticode) [reply swap/protocol_test.go]: Thanks!

**Review by holisticode — APPROVED** [review]
Builds failed but it seems unrelated.
Let's wait until core is aware of these failures (I already reported) and then restart the builds before merging, better to have had green builds before merging.
Great job.

**Pull request merged**